docs/syntax_by_example.md

```diff
@@ -0,0 +1,367 @@
+# Evy Syntax by Example
+
+The following examples on various aspects of `evy`'s syntax give an
+intuitive understanding of the language. For a formal language
+specification see the [syntax grammar].
+
+[syntax grammar]: syntax_grammar.md
+
+## Comment
+
+    // This is comment
+
+## Declaration
+
+    x:num     // declaration: num, string, bool, any, [] {}
+    y := 1    // declaration through type inference (num)
+
+## Assignment
+
+    z = 5
+
+## Expression
+
+    x := 5 * (x + z)  - 2 / 7.6           // arithmetic number expression
+    b := !trace and debug or level == ""  // bool expressions
+
+## Strings
+
+    s1 := "quotation mark : \" " // escaping
+    s2 := "abc" + "123"          // concatenation
+    s3 := "newline: \n indentation: \t"
+    s4 := s2[0]                  // "a"
+    s5 := s2[1:4]                // "bc"
```

Comment:
it would be good to have an example with multi-byte chars so you can see strings are indexed by character not byte.

Reply:
done.